### PR TITLE
Add matmul kernel test for CPU backend

### PR DESCRIPTION
**Pull request by nhat-nguyen**
Add the latest matmul triton tutorial as a test for the CPU backend. This exposed a bug in the TritonToLinalg pass where we need to initialize the init tensor with all zeroes first since allocated buffers aren't guaranteed to be zero-initialized.

### CODE REVIEW

File: python/examples/test_matmul.py
Comment (manbearian) [inline python/examples/test_matmul.py:1]: i was hoping to avoid replicating the entire tutorial code. Is there a way we can import this?

**Review by manbearian — APPROVED** [review]
Looks good, my only feedback is around looking for a way to avoid replicating the existing tutorial code.

File: python/examples/test_matmul.py
Comment (nhat-nguyen) [reply python/examples/test_matmul.py:1]: yea this turns out harder than i expected:
1) we need to modify the device type from `cuda` to `cpu` in the original code anyway
2) our project is independent from the tutorial code which lives in the `triton` repo, so we can't directly access those files anyway

**Comment by manbearian** [comment]
Yes, no need to fix here... but we really need a way to address this... I wonder if we can modify the tutorials themselves with some conditional code to check if CPU is supported and test CPU back-end if it is?

ian Bearman
Principal Software Engineering Manager
AI Frameworks: Compilation and Optimization (Maia 100<https://azure.microsoft.com/en-us/blog/microsoft-azure-delivers-purpose-built-cloud-infrastructure-in-the-era-of-ai/> and beyond!)
Now hiring!<https://aka.ms/ianb.jobs>
#BlackLivesMatter
/* Making your AI faster, cheaper, smarter! */

**Pull request merged**